vanilla-tests/src/main/java/fixtures/inheritance/passdiscriminator/models/MetricAlertCriteria.java

```diff
@@ -0,0 +1,70 @@
+package fixtures.inheritance.passdiscriminator.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.core.annotation.JsonFlatten;
+import com.fasterxml.jackson.annotation.JsonAnyGetter;
+import com.fasterxml.jackson.annotation.JsonAnySetter;
+import com.fasterxml.jackson.annotation.JsonIgnore;
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.annotation.JsonTypeInfo;
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import java.util.HashMap;
+import java.util.Map;
+
+/** The rule criteria that defines the conditions of the alert rule. */
+@JsonTypeInfo(
+        use = JsonTypeInfo.Id.NAME,
+        include = JsonTypeInfo.As.EXISTING_PROPERTY,
+        property = "odata\\.type",
```

Comment:
`odata.type` does not appear in this class. I guess you need a few more place to change in `ModelTemplate`.

Reply:
Hmm, yes this is a good question on whether the configuration both passes the type to the child class and has the child class generate a field for the type or whether this simply just passes the type to the child class.

Reply:
I probably misunderstand the purpose of the feature. The property/field does not intended to be present in the class.
So forget my previous comment.

From this section in issue, it seems parent class should have ``include = JsonTypeInfo.As.PROPERTY` instead of `include = JsonTypeInfo.As.EXISTING_PROPERTY`?

```
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "@odata.type", defaultImpl = TokenFilter.class, visible = true)
@JsonTypeName("TokenFilter")
@JsonSubTypes({
    @JsonSubTypes.Type(name = "#Microsoft.Azure.Search.AsciiFoldingTokenFilter", value = AsciiFoldingTokenFilter.class)
    /*
      * Additional sub-type odata.type value and class mappings here
      */
})
@Fluent
public class TokenFilter;


@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.EXISTING_PROPERTY, property = "@odata.type", visible = true)
@JsonTypeName("#Microsoft.Azure.Search.AsciiFoldingTokenFilter")
@Fluent
public final class AsciiFoldingTokenFilter extends TokenFilter;
```